public/app/plugins/panel/heatmap-new/ColorScale.tsx

```diff
@@ -10,55 +10,72 @@ type Props = {
 
   // Show a value as string -- when not defined, the raw values will not be shown
   display?: (v: number) => string;
+  hoverValue?: number;
 };
 
 type HoverState = {
   isShown: boolean;
   value: number;
 };
 
-export const ColorScale = ({ colorPalette, min, max, display }: Props) => {
+export const ColorScale = ({ colorPalette, min, max, display, hoverValue }: Props) => {
   const [colors, setColors] = useState<string[]>([]);
-  const [hover, setHover] = useState<HoverState>({ isShown: false, value: 0 });
-  const [cursor, setCursor] = useState({ clientX: 0, clientY: 0 });
+  const [scaleHover, setScaleHover] = useState<HoverState>({ isShown: false, value: 0 });
+  const [percent, setPercent] = useState<number | null>(null);
+  const [eventOffsetWidth, setEventOffsetWidth] = useState<number>(0);
+
+  const theme = useTheme2();
+  const styles = getStyles(theme, colors);
 
   useEffect(() => {
     setColors(getGradientStops({ colorArray: colorPalette }));
   }, [colorPalette]);
 
-  const theme = useTheme2();
-  const styles = getStyles(theme, colors);
+  const offsetWidth = useMemo(() => eventOffsetWidth, [eventOffsetWidth]);
 
   const onScaleMouseMove = (event: React.MouseEvent<HTMLDivElement>) => {
+    setEventOffsetWidth((event.target as any).offsetWidth as number);
```

Comment:
Lets just use `(event.target as any).offsetWidth` directly in this PR

The memoization strategy used here does not actually help... since the expensive call happens every time regarless